test/db_connection/metrics_test.exs

```diff
@@ -0,0 +1,126 @@
+defmodule MetricsTest do
+  use ExUnit.Case, async: false
+
+  defmodule DummyDB do
```

Comment:
We already have a DummyDB that we use in other tests (via the agent+connection). Could we perhaps re-use that instead?

Reply:
Hm, I figured I'd just reimplement a smaller DummyDB here for maximum clarity and to segregate the tests from the other ones.